futures-util/src/future/map.rs

```diff
@@ -1,48 +1,44 @@
 use core::marker::Unpin;
 use core::mem::PinMut;
-
-use futures_core::{Future, Poll};
-use futures_core::task;
+use futures_core::future::Future;
+use futures_core::task::{Context, Poll};
 
 /// Future for the `map` combinator, changing the type of a future.
 ///
 /// This is created by the `Future::map` method.
 #[derive(Debug)]
 #[must_use = "futures do nothing unless polled"]
-pub struct Map<A, F> where A: Future {
-    future: A,
-    f: Option<F>,
+pub struct Map<Fut, F> {
+    future: Fut,
+    op: Option<F>,
 }
 
-pub fn new<A, F>(future: A, f: F) -> Map<A, F>
-    where A: Future,
-{
-    Map {
-        future,
-        f: Some(f),
-    }
-}
+impl<Fut, F> Map<Fut, F> {
+    unsafe_pinned!(future -> Fut);
+    unsafe_unpinned!(op -> Option<F>);
 
-impl<A: Future, F> Map<A, F> {
-    unsafe_pinned!(future -> A);
-    unsafe_unpinned!(f -> Option<F>);
+    /// Creates a new Map.
+    pub(super) fn new(future: Fut, op: F) -> Map<Fut, F> {
```

Comment:
Why change these functions to `pub(super)` methods in some places and not others? I'm fine with it either way, but we should be consistent.